arch/arc/include/swap_macros.h

```diff
@@ -86,6 +89,9 @@ extern "C" {
 	ld sp, [r2, _thread_offset_to_sp]
 
 #ifdef CONFIG_FP_SHARING
+	ld r13, [r2, ___thread_base_t_user_options_OFFSET]
```

Comment:
see also comments in the test code. I think r58/r59 need to be part of the context that's unconditionally saved/restored, if present.

Reply:
We do need to save/restore `ACCL`/`ACCH` registers all the time BUT only when they exist.
And they exist only if `MPY_OPTION > 6` or `FPU_FMA_OPTION`.

So in case of the Linux kernel we rely on user specified availability of `ACCL`/`ACCH` with run-time check if user setting matches real HW. That way we don’t need to check HW properties on every context save/restore, see:
1. https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/arc/Kconfig#n404
2. https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/arc/include/asm/entry-arcv2.h#n48

Reply:
already fixed